### DISCUSSION TITLE
iter_patch memory consumption

### DISCUSSION THREAD

**Discussion by schellenchris** [Project-MONAI/MONAI#5610] [General]
Hello,

I was playing around with the GridPatchDataset / training with smaller Slices than the original volume, when I ran into some weird RAM OOM problems. After some investigation i noticed that line 290 in data/utils.py cause the problem:
https://github.com/Project-MONAI/MONAI/blob/d0db5fd9d4da3bc7027d22c52bb9c405c3b9e879/monai/data/utils.py#L288-L290

is there a reason to pad the image that much (especially padding the channels doesn't make sense for me)?
I managed to reduce the memory by 15! GB by modifying the code a bit and just padding the axis slices are generated from. I was just wondering if I am missing out on something.

**Comment by wyli** [comment]
thanks for reporting, I create an issue to track it https://github.com/Project-MONAI/MONAI/issues/5611

**Reply by wyli** [reply]
this is addressed by https://github.com/Project-MONAI/MONAI/pull/5612 by no padding when the corresponding dimension of the patch size is set to `None` or 0.

**Reply by schellenchris** [reply]
works like a charm, no memory problems anymore. thank you!
